fitbenchmarking/fitting/controllers/mantid_controller.py

```diff
@@ -0,0 +1,98 @@
+"""
+Implements a controller for the Mantid fitting software.
+"""
+
+from mantid import simpleapi as msapi
+from mantid.api import *
+from mantid.fitfunctions import *
+import numpy as np
+
+from fitbenchmarking.fitting.controllers.base_controller import Controller
+
+
+class MantidController(Controller):
+    """
+    Controller for the Mantid fitting software.
+    """
+
+    def __init__(self, problem, use_errors):
+        """
+        Setup workspace, cost_function, ignore_invalid, and initialise vars
+        used for temporary storage within the mantid controller
+        """
+        super(MantidController, self).__init__(problem, use_errors)
+
+        self._param_names = [row[0] for row in problem.starting_values]
+
+        self._cost_function = 'Least squares' if use_errors \
+            else 'Unweighted least squares'
+
+        self._ignore_invalid = use_errors & ('WISH17701' not in problem.name)
```

Comment:
Please explain what _ignore_invalid is. Also noticed it is not defined in the base class - should it be? Here also explain the reason for the hardcoded 'WISH17701' not in problem.name

Reply:
It is something I copied from the old implementation.
```
def get_ignore_invalid(problem, cost_function):
    """
    Helper function that sets the whether the mantid fitting software
    ignores invalid data or not. This depends on the cost function.

    @param prob :: object holding the problem information
    @param const_function :: cost function used in fitting

    @returns :: boolean depending on whether to ignore invalid data or not
    """

    ignore_invalid = cost_function == 'Least squares'

    # The WISH data presents some issues
    # For which this ad-hoc if must is present
    if 'WISH17701' in problem.name:
        ignore_invalid = False

    return ignore_invalid
```

I was trying not to change the actual functionality so copied it accross (although I should probably have commented it

Reply:
There's some discussion of this on PR #109 (and it's parent issue).  Do you think we should reopen #38?

Reply:
Looks like it works without. I'll remove `_ignore_invalid`

Reply:
I'll leave it as using the "ignoreInvalidData" option in Mantid though based on whether we are using errors or not as I don't know why it's there/ what it does. If ou don't think it's needed we could put the investigation into a new issue.